### PR TITLE
Led integration

### PR DESCRIPTION
**Pull request by SebZanardo**
Currently climbing has no status lights as more logic needs to be added to stop race conditions and prioritise certain status's over others.

### CODE REVIEW

**Comment by james-ward** [comment]
Can this be simulated?

**Comment by SebZanardo** [comment]
> Can this be simulated?

Yes but you can't trigger the statemachine that sets the led status's without a controller

**Comment by james-ward** [comment]
I have just run this in the sim fine. I don't think the logic is correct.

1. Press intake button -> LEDs solid purple. Flashing is better - this is a high risk situation wrt to impacts from other robots.
2. Make state machine think it has a note by toggling DIO 2 low. Lights flash light blue? Should be solid green if we are in range, red if not in range.
3. Press trigger to fire -> robot rotates to goal and LEDs go solid red for a moment, then solid green. Should remain green when shooting - don't give other teams notice that we are in the process of shooting.
4. Abort "shooting" (DIO 2 still low) and drive to other side of field. LEDs remain green - should go red.
5. "Shoot" (DIO 2 high) - LEDs flash white. LEDs should be turned off.

> Yes but you can't trigger the statemachine that sets the led status's without a controller

I don't understand - everything worked fine for me.

File: controllers/shooter.py [hunk 1]
```diff
@@ -37,14 +39,13 @@ def aiming(self, initial_call) -> None:
...
             self.aim()
         else:
-            if (
-                self.chassis.at_desired_heading()
-                and self.shooter_component.is_ready()
-                and self.in_range()
-            ):
-                self.next_state(self.firing)
+            if self.chassis.at_desired_heading():
+                self.status_lights.in_range()
```
Comment (james-ward) [inline controllers/shooter.py:43]: Move this check to the `note.py` state machine in the `holding_note` state. At the moment the status lights only get updated when the command to fire is given. They should be updating all the time a note is held.

**Comment by james-ward** [comment]
The intake, in range, out of range look good to me.

**Review by james-ward — APPROVED** [review]
Tops!

**Pull request merged**